backend/common/src/main/java/ai/verta/modeldb/common/GracefulShutdown.java

```diff
@@ -27,38 +27,41 @@ public void customize(Connector connector) {
 
   @Override
   public void onApplicationEvent(ContextClosedEvent event) {
+    LOGGER.info("GracefulShutdown starting for Spring server");
     this.connector.pause();
     var executor = this.connector.getProtocolHandler().getExecutor();
     if (executor instanceof ThreadPoolExecutor) {
       try {
         var threadPoolExecutor = (ThreadPoolExecutor) executor;
-        int nfsActiveRequestCount = threadPoolExecutor.getActiveCount();
+        int springServerActiveRequestCount = threadPoolExecutor.getActiveCount();
 
-        while (nfsActiveRequestCount > 0) {
-          nfsActiveRequestCount = threadPoolExecutor.getActiveCount();
-          LOGGER.info("NFS Active Request Count in while: {}", nfsActiveRequestCount);
+        while (springServerActiveRequestCount > 0) {
+          springServerActiveRequestCount = threadPoolExecutor.getActiveCount();
+          LOGGER.info(
+              "Spring server active Request Count in while: {}", springServerActiveRequestCount);
```

Comment:
We should call `shutdown()` first, [that will stop new requests from being accepted](https://tomcat.apache.org/tomcat-8.5-doc/api/org/apache/tomcat/util/threads/ThreadPoolExecutor.html#shutdown()).

Also, this while loop here here has no timeout, it will just keep going forever.